eras/allegra/impl/testlib/Test/Cardano/Ledger/Allegra/Imp/UtxowSpec.hs

```diff
@@ -0,0 +1,100 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedLists #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Test.Cardano.Ledger.Allegra.Imp.UtxowSpec (
+  spec,
+  specAllegra,
+  genInvalidMetadatum,
+) where
+
+import Cardano.Ledger.Allegra.Scripts (Timelock (..))
+import Cardano.Ledger.Allegra.TxAuxData
+import Cardano.Ledger.BaseTypes
+import Cardano.Ledger.Core
+import Cardano.Ledger.Keys (coerceKeyRole)
+import Cardano.Ledger.SafeHash (HashAnnotated (..))
+import Cardano.Ledger.Shelley.Rules (ShelleyUtxowPredFailure (..))
+import Cardano.Ledger.Shelley.TxAuxData (Metadatum (..))
+import qualified Data.ByteString.Char8 as BS
+import qualified Data.Map as Map
+import qualified Data.Text as T
+import Data.Word (Word64)
+import Lens.Micro
+import Test.Cardano.Ledger.Allegra.ImpTest (produceScript)
+import Test.Cardano.Ledger.Core.KeyPair (mkWitnessVKey)
+import Test.Cardano.Ledger.Imp.Common
+import Test.Cardano.Ledger.Shelley.ImpTest
+
+spec ::
+  forall era.
+  ( ShelleyEraImp era
+  , NativeScript era ~ Timelock era
+  , InjectRuleFailure "LEDGER" ShelleyUtxowPredFailure era
+  ) =>
+  SpecWith (ImpTestState era)
+spec = describe "UTXOW" $ do
+  it "MissingScriptWitnessesUTXOW" $ do
```

Comment:
Both `MissingScriptWitnessesUTXOW` and `ExtraneousScriptWitnessesUTXOW` are applicable to `Shelley` as well, thus they can be moved into the Shelley imp spec instead.

Also once you do that we can remove this test:
* https://github.com/IntersectMBO/cardano-ledger/blob/2c9ba10d085331fa99840a63b40b8e8b716fba5d/eras/shelley/test-suite/test/Test/Cardano/Ledger/Shelley/RulesTests.hs#L169-L183
* https://github.com/IntersectMBO/cardano-ledger/blob/2c9ba10d085331fa99840a63b40b8e8b716fba5d/eras/shelley/test-suite/test/Test/Cardano/Ledger/Shelley/RulesTests.hs#L94